README.md

```diff
@@ -32,10 +32,10 @@ This library provides auto-renewed tokens for Google service authentication.<br>
 use google_authz::{Credentials, GoogleAuthz};
 
 let credentials = Credentials::builder().build().await.unwrap();
-let service = GoogleAuthz::builder(service).credentials(credentials).build().await;
+let service = GoogleAuthz::builder(service).credentials(credentials).build().await.unwrap();
```

Comment:
I think changing the public API is the right thing to do, because the `GoogleAuthz` builder truly is fallible. Hiding the fallibility using internal unwraps doesn't really help anybody. The compiler will naturally help anyone moving to the new version of the crate. 

My philosophy here is, in general, I feel fallible functions should return `Result` instead of hiding their fallibility using `unwrap`. I personally am only okay with `unwrap` when, within a function or in its immediate context, we've validated the necessary logical invariants. 

do you mean to validate the RSA key while building the `Credentials` instead of when building the `GoogleAuthz`? I'm open to that, it's slightly more code change, but it makes the error be caught earlier. I don't necessarily think one way is better than the other